app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -26,31 +31,60 @@ class AgendaRepositoryImpl(
+    override suspend fun syncAgendaItems() {
+        reminderRepository.uploadCreateAndUpdateModifiedReminders()
+
         val reminderDeleteIds = db.getReminderDao().getModifiedReminders().filter {
             it.modifiedType == ModifiedType.DELETE
         }.map {
             it.id
         }
         val syncAgendaRequest = SyncAgendaRequest(emptyList(), emptyList(), reminderDeleteIds)
-        return getAuthResult { api.syncAgendaItems(syncAgendaRequest) }
+        when (getAuthResult { agendaApi.syncAgendaItems(syncAgendaRequest) }) {
+            is AuthResult.Authorized -> {
+                reminderDeleteIds.map {
+                    db.getReminderDao().deleteModifiedReminderById(it)
+                }
+            }
+            is AuthResult.Unauthorized -> Log.e(
+                "SyncAgendaItem",
+                "Unauthorized, could not run agendaApi.syncAgendaItems(syncAgendaRequest)"
+            )
+            is AuthResult.UnknownError -> Log.e(
+                "SyncAgendaItem",
+                "Unknown Error, could not run agendaApi.syncAgendaItems(syncAgendaRequest)"
+            )
+        }
     }
 
-    override suspend fun updateAgendaOfDate(time: Long, timezone: String) {
+    override suspend fun updateAgendaOfDate(date: LocalDate) {
+        syncAgendaItems()
         when (
             val results = getAuthResult {
-                api.getAgendaItems(
-                    timezone,
-                    time
+                agendaApi.getAgendaItems(
+                    timezone = TimeZone.getDefault().id,
+                    time = date.atTime(LocalTime.now())
+                        .atZone(TimeZone.getDefault().toZoneId()).toEpochSecond()
                 )
             }
         ) {
             is AuthResult.Authorized -> {
-                val remoteReminders = results.data?.reminders?.map { reminderDto ->
-                    reminderDto.toReminder()
+                val localReminders = db.getReminderDao().getOneTimeListOfReminders()
+                localReminders.map { localReminder ->
+                    val remoteReminderIds = results.data?.reminders?.map { it.id } ?: emptyList()
+                    if (!remoteReminderIds.contains(localReminder.id)) {
+                        db.getReminderDao().deleteReminder(localReminder)
+                    }
                 }
-                remoteReminders?.map { reminder ->
-                    db.getReminderDao().upsertReminder(reminder.toReminderEntity())
+                results.data?.reminders?.map { reminderDto ->
+                    val localReminder = db.getReminderDao().getReminderById(reminderDto.id)
+                    db.getReminderDao().upsertReminder(
+                        reminderDto.toReminderEntity(
+                            // Android Studio says local reminder will never equal null,
+                            // but can if reminder doesn't exist in db
+                            isDone = localReminder.isDone ?: false
```

Comment:
Android Studio is actually correct here I think. Is the return type of `getReminderById()` `Reminder?` or `Reminder`? If it's the latter, the function will throw an exception if the reminder doesn't exist.